src/App.js

```diff
@@ -1,25 +1,56 @@
-import logo from './logo.svg';
-import './App.css';
-
-function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+import React, { Component } from 'react';
+import axios from 'axios';
+import Form from './Form.js';
+
+// https://us1.locationiq.com/v1/reverse.php?key=YOUR_ACCESS_TOKEN&lat=LATITUDE&lon=LONGITUDE&format=json
+
+export default class App extends Component {
+
+  constructor(props) {
+    super(props);
+    this.state ={
+      queryCity:'',
+      locationObject: {},
+      error: false
+    }
+  }
+
+ 
+ 
+getlocation = async() => {
+
+    try {
+    let result = await axios.get(`https://us1.locationiq.com/v1/search.php?key=${process.env.REACT_APP_API_KEY}&q=${this.state.queryCity}&format=json`)
+    this.setState({ locationObject: result.data[0] })
+    } catch (error) {
+      console.error(error);
+      this.setState({error: true});
+    }
+  }
+
+handleSubmit = (event) => {
+  event.preventDefault();
+  this.setState({queryCity: event.target.city.value }, this.getlocation);
 }
 
-export default App;
+  render() {
+    return (
+      <div>
+        
+        <form onSubmit ={this.handleSubmit}>
+          <input type="text" placeholder="Enter City Name Here" name="city"/>
+          <button type="submit">Explore!</button>
+        </form>
+
+        {this.state.locationObject.display_name ? <p>{this.state.locationObject.display_name}, Earth <br></br> <br></br> Latitude: {this.state.locationObject.lat} Longitude: {this.state.locationObject.lon}</p> : <p>Search for a city.</p>}
+        <img src={`https://maps.locationiq.com/v3/staticmap?key=${process.env.REACT_APP_API_KEY}&center=${this.state.locationObject.lat},${this.state.locationObject.lon}&zoom=10&size=400x400&format=png&maptype=roadmap&markers=icon:<icon>|<latitude>,<longitude>&markers=icon:<icon>|<latitude>,<longitude>`} alt="map"/>
+        {this.state.error && <p>There was an error with your request.</p>}
+        <Form>
+
+        </Form >
```

Comment:
- The lab asks to show data using Bootstrap components. See if you can hook up your Form component you started on. 👍